clarifai_model_utils/llm_eval/evaluator/harness_eval/Readme.md

```diff
@@ -0,0 +1,46 @@
+# Module description
+
+### 1. [evaluate.py](evaluate.py)
+
+This module is used to execute the evaluation process using `harness-eval` as the backend with customized templates listed in the [template folder](./template/). Each template is structured following the harness template (or Task) as described in the [documentation](https://github.com/EleutherAI/lm-evaluation-harness/blob/main/docs/task_guide.md).
+
+The `evaluate.py` module serves as a wrapper over [the harness implementation](https://github.com/EleutherAI/lm-evaluation-harness/blob/main/lm_eval/evaluator.py), forcing the use of the Clarifai model and allowing template parameters to be set via method arguments, which is more convenient compared to setting them via a YAML file. This approach is designed to save users from having to read through harness task configuration details, while developers can use it to maintain or expand the codebase.
+
+Other features include:
+- Retrieving metric names per template
+- Fetching documentation per template
+- Accessing configuration per template
+
+### 2. [llm.py](llm.py)
+
+This file contains the implementation of the `Clarifai SDK model/workflow` derived from the harness base model for use in evaluation. It includes two ready components:
+- Text model
+- RAG workflow
+
+### 3. [judge_llm.py](judge_llm.py)
+
+The `judge_llm.py` module contains the `JudgeLLM` class whose `process` method is inserted into the `process_results` field of templates `llm_as_judge` and `rag`. This allows any judge model to be chosen by providing its URL. You can refer to its usage in [evaluate.py](https://github.com/Clarifai/clarifai-model-utils/blob/09eb2ab5f5b3a4db898dbb90c4f8de82ad746f4c/clarifai_model_utils/llm_eval/evaluator/harness_eval/evaluate.py#L241).
```

Comment:
I noticed `process_rag_result` in that judge_llm.py as well. Is that meant for workflows? Or the RAG workflow specifically?

Reply:
I think what would make this more clear is to include, given a new model or a new workflow, what the steps are (what methods to create where) to enable evaluation, including saving options if available.